### ISSUE TITLE
add plugin.Classer interface

### ISSUE THREAD

**Issue by miekg** [coredns/coredns#1499]
core/dnsserver/server.go has some whacky code that unblock CH class queries if either proxy, chaos or (new forward) is loaded. This is done by `Name()` on the plugin.

It is nicer to implement a plugin.Classer interface:
~~~
type Classer interface {
      Class() []uint16
}
~~~
That specifies what classes should be *unblocked* (excluding `IN` which is always allowed). In the current use case only `CH` makes sense (and I think it is the only one in use on the net anyway). So are short impl. cloud by:
~~~
type Chaoser interface {
     Chaos() bool
}
~~~
If a plugin implements it we will open the CH class

**Comment by miekg** [comment]
This is a nice-to-have, but noone is working on it, and so few plugins need this that a separate interface might be overkill. Closing for now.
